### ISSUE TITLE
"You are not logged in" followed by "cx_freeze" crash (1.2.0 Win10)

### ISSUE THREAD

**Issue by Dustie** [Nick-Gottschlich/Social-Amnesia#47]
Trying to deleted Reddit comments I get "You are not logged in" even though it does show amount of comments. The popup windows is empty. Closing it down and launching it again I get the following error:

https://i.imgur.com/5UeCrRU.png

I've tried deleting the reddit_state and both the .config and .SocialAmnesia folders but at launch I then get the "not logged in" followed by the same crash. I have changed no settings except filling out the login details.

After trying several times yesterday it suddenly worked and it deleted some comments but it was both really slow and it had to be restarted more than once. I let it run overnight and today it is totally dead. I can't find any logfiles or config files so not sure how much help this is.

**Comment by Nick-Gottschlich** [comment]
Whoops, didn't mean to close that. One thing you can try is to look for a file called "reddit_state.db" on your computer. If you delete that it should wipe all of the saved data in Social Amnesia. Then open up the app again and let me know what happens.

If that doesn't fix your issues let me know and I can debug further.

Thank you for opening the issue!

**Comment by Dustie** [comment]
As mentioned I already tried deleting reddit_state.db.dat and reddit_state.db.dir (and the whole folder). I could not find any files only named reddit_state.db via a Windows search. Deleting those files does reset everything but I still get the not logged in. As you can see it does say I am logged in:

https://i.imgur.com/jxP2tcc.png

(I think I found another bug just now. I missed a letter in my password and the program then opens a reddit page (Oauth?) with a "bad request". Let me know if I should open another issue.)

**Comment by Nick-Gottschlich** [comment]
Ah I completely missed where you said you deleted reddit_state, that's my bad 😅.

So, just to confirm, when you click "delete comments" and then hit "proceed" you see the error saying "You are not logged in."?

If so, that's the translation from `<class 'KeyError'>`, which indicates something is wrong with the reddit_state at that point =/. 

One thing I can try to do when I get a chance is confirm I actually built the Windows 1.2.0 release properly, I'm hoping that might be the problem.

**Comment by Dustie** [comment]
Eh almost. I should have made a better bug report. Let me try again with the steps I take :)
After deleting .config and .SocialAmnesia:


- Launch program (as admin or not makes no difference)
- Fill out username, password, client id and secret
- Click "Login to reddit"
- "Logged in to reddit as <username>"
- In the reddit tab click "Delete comments"
- The "The following comments will be deleted/edited" window pops up (empty)
- Then an Error box with "You are not logged in!"
- Under "Deletion" it says "0/192 items processed so far" so it does connect
- Click OK in the error message


Hmm. I think there are actually two bugs here. The login issue and the cx_freeze crash seems unrelated. I'm not getting the cx_freeze crash now but if I change the max score and/or close the *main window* before the ""The following comments will be deleted" window I do get the crash. I'm not sure which of the two causes it since I for some reason don't get it every time I reset the app. Maybe something goes wrong when the secondary window is closed after main?

**Comment by Dustie** [comment]
(Sorry about the spam)

It now gets stuck at item 100 but unlike in issue #37 I don't use two factor auth. Clicking Delete again and then Proceed fixes it and it deletes like it should.

I'm a (newbie) programmer myself and I'm happy I don't have to make sense of this thread/bug. Let me know if I should test something.

**Comment by Nick-Gottschlich** [comment]
Closing this out now that Version 2 of Social Amnesia has been released.
